### ISSUE TITLE
Creating sink throws compile error: "No type named 'vector' in namespace 'std'" in formatter.h

### ISSUE THREAD

**Issue by winduptoy** [gabime/spdlog#181]
When I create my own sink, I get a strange compile error _"No type named 'vector' in namespace 'std'"_ and directs me to **formatter.h**. Here's my sink:

```c++
#include <memory>
#include <spdlog/sinks/sink.h>

#include "AppLog.hpp"

class AppLogSink : public spdlog::sinks::sink {

public:
	
	std::shared_ptr<AppLog> mpLogGUI;
	AppLogSink(std::shared_ptr<AppLog> logGUI) : mpLogGUI(logGUI) {}
	
	void log(const spdlog::details::log_msg & msg) override {
		// Do my thing
	}
	
	void flush() override {
		// Do my thing
	}
};
```

Slapping an `#include <vector>` at the top of **formatter.h** fixes it. Is this worthy of a PR or am I doing something wrong?

**Comment by gabime** [comment]
formatter.h is comes from an external lib called  cppformat. please report in its repo.
Anyway, I think that including  spdlog.h at top of your sink will solve this.

**Comment by vitaut** [comment]
@gabime, `formatter.h` is not a part of cppformat, `format.h` is. Looks like

```c++
#include <vector>
```
is missing in `formatter.h` because it uses `std::vector` in https://github.com/gabime/spdlog/blob/master/include/spdlog/formatter.h#L33

**Comment by winduptoy** [comment]
@gabime Including spdlog.h does not fix it.

**Comment by gabime** [comment]
@vitaut right. sorry,i confused between them

**Comment by winduptoy** [comment]
@vitaut Were you able to confirm that this happens on your end as well?

**Comment by vitaut** [comment]
@winduptoy I haven't tried to reproduce the problem if you mean that but I'm pretty sure that the cause is missing include in `formatter.h` as I mentioned in the comment above.

**Comment by godbyk** [comment]
This should've been fixed in master by PR #182. Is anyone still experiencing this problem?

**Comment by gabime** [comment]
Fixed in PR #182
